code/modules/projectiles/guns/projectile/revolver.dm

```diff
@@ -233,3 +233,34 @@
 
 /obj/item/weapon/gun/projectile/russian/empty/New()
 	update_icon()
+
+/obj/item/weapon/gun/projectile/colt
+	name = "\improper Colt Single Action Army"
+	desc = "The greatest handgun ever made."
+	icon_state = "colt"
+	max_shells = 6
+	var/cocked = FALSE
+
+/obj/item/weapon/gun/projectile/colt/update_icon()
+	if(cocked)
+		icon_state = "colt_cocked"
+	else
+		icon_state = "colt"
+
+/obj/item/weapon/gun/projectile/colt/attack_self(mob/user, var/callparent = FALSE)
+	if(callparent)
+		return ..(user)
+	if(!cocked)
+		cocked = TRUE
+		update_icon()
+		to_chat(user, "You cock \the [src].")
```

Comment:
I ran out of ways to interact with the revolver.
I feel like using a verb would be less than optimal.